grants list caused panic

**Describe the bug**

```
mchiang@Michaels-MacBook-Pro ~ % infra grants list
panic: runtime error: slice bounds out of range [2:0]

goroutine 1 [running]:
github.com/infrahq/infra/uid.PolymorphicID.ID({0x0?, 0x0?})
	/home/runner/work/infra/infra/uid/polymorphic.go:16 +0x5c
github.com/infrahq/infra/internal/cmd.listInfo(0x140004a9900, {0x6bc0a1c8952000, 0x623df029, 0x1, 0x623df029, {0x0, 0x0}, {0x140003f4d08, 0x4}, {0x14000443a80, ...}, ...})
	/home/runner/work/infra/infra/internal/cmd/list.go:121 +0x64
github.com/infrahq/infra/internal/cmd.newGrantsListCmd.func1(0x14000822000?, {0x105c0a780?, 0x0?, 0x0?})
	/home/runner/work/infra/infra/internal/cmd/grants.go:57 +0x230
github.com/spf13/cobra.(*Command).execute(0x14000822000, {0x105c0a780, 0x0, 0x0})
	/home/runner/go/pkg/mod/github.com/infrahq/cobra@v1.4.0-groups/command.go:883 +0x4c4
github.com/spf13/cobra.(*Command).ExecuteC(0x140001d2b00)
	/home/runner/go/pkg/mod/github.com/infrahq/cobra@v1.4.0-groups/command.go:1001 +0x354
github.com/spf13/cobra.(*Command).Execute(...)
	/home/runner/go/pkg/mod/github.com/infrahq/cobra@v1.4.0-groups/command.go:929
github.com/infrahq/infra/internal/cmd.Run()
	/home/runner/work/infra/infra/internal/cmd/cmd.go:615 +0x40
main.main()
	/home/runner/work/infra/infra/main.go:10 +0x20
```

infra v0.7.0

--- a/uid/polymorphic.go
+++ b/uid/polymorphic.go
@@ -13,6 +13,9 @@ func (p PolymorphicID) String() string {
 }
 
 func (p PolymorphicID) ID() (ID, error) {
+	if len(p) < 2 {
+		return ID(0), fmt.Errorf("invalid polymorphic ID encountered: %v", p)
+	}
 	return ParseString(string(p)[2:])
 }
 

